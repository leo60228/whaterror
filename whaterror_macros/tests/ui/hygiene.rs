mod local_hygiene {
    mod whaterror {
        pub const CONSTANT: i32 = 5;
    }

    #[whaterror_macros::whaterror(|| { assert_eq!(whaterror::CONSTANT, 5); let _ = inner; let _ = handle; })]
    fn local_hygiene() {
        let _ = inner;
        let _ = handle;
        let _ = whaterror::CONSTANT;
    }
}

mod naming_conflicts {
    #[whaterror_macros::whaterror(())]
    fn whaterror() {}

    #[whaterror_macros::whaterror(())]
    fn inner() {}

    #[whaterror_macros::whaterror(())]
    fn handle() {}

    #[whaterror_macros::whaterror(())]
    fn thunk() {}
}

fn main() {}
