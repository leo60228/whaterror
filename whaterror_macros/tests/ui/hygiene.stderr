error[E0425]: cannot find value `inner` in this scope
 --> $DIR/hygiene.rs:8:17
  |
8 |         let _ = inner;
  |                 ^^^^^ not found in this scope
  |
help: consider importing this function
  |
2 |     use crate::naming_conflicts::inner;
  |

error[E0425]: cannot find value `handle` in this scope
 --> $DIR/hygiene.rs:9:17
  |
9 |         let _ = handle;
  |                 ^^^^^^ not found in this scope
  |
help: consider importing this function
  |
2 |     use crate::naming_conflicts::handle;
  |

error[E0425]: cannot find value `inner` in this scope
 --> $DIR/hygiene.rs:6:84
  |
6 |     #[whaterror_macros::whaterror(|| { assert_eq!(whaterror::CONSTANT, 5); let _ = inner; let _ = handle; })]
  |                                                                                    ^^^^^ not found in this scope
  |
help: consider importing this function
  |
2 |     use crate::naming_conflicts::inner;
  |

error[E0425]: cannot find value `handle` in this scope
 --> $DIR/hygiene.rs:6:99
  |
6 |     #[whaterror_macros::whaterror(|| { assert_eq!(whaterror::CONSTANT, 5); let _ = inner; let _ = handle; })]
  |                                                                                                   ^^^^^^ not found in this scope
  |
help: consider importing this function
  |
2 |     use crate::naming_conflicts::handle;
  |
